src/openforms/formio/dynamic_config/date.py

```diff
@@ -113,13 +113,16 @@ def calculate_delta(
         assert config["mode"] == "relativeToVariable"
 
         base_value = cast(
-            Optional[datetime],
+            Optional[Union[datetime, str]],
             glom(data, config["variable"], default=None),
         )
-        # can't do calculations on values that don't exist
-        if base_value is None:
+        # can't do calculations on values that don't exist or are empty
+        if not base_value:
             return None
 
+        # if it's not empty-ish, it's a datetime
+        base_value = cast(datetime, base_value)
+
```

Comment:
Had a failing test because the variable default value was a string for type datetime (in case no value is set yet), which can happen, so this is now handled and caught too.